lib/tokenize.cpp

```diff
@@ -4053,7 +4053,7 @@ void Tokenizer::setVarIdPass1()
             }
 
             try { /* Ticket #8151 */
-                decl = setVarIdParseDeclaration(&tok2, variableMap, scopeStack.top().isExecutable, isCPP(), isC());
+                decl = setVarIdParseDeclaration(&tok2, variableMap, scopeStack.top().isExecutable, isCPP(), isC(), isHeader());
```

Comment:
I don't have a good feeling for this isHeader(). you are not supposed to check a header directly:

    cppcheck file.h

Reply:
People do that and it will happen with IDE integrations. There's also a push to improve handling of static analysis of header within clang-tidy and CMake.